modules/javamail/src/main/scala/emil/javamail/internal/ops/ListFoldersRecursive.scala

```diff
@@ -0,0 +1,36 @@
+package emil.javamail.internal.ops
+
+import cats.effect.Sync
+import emil.javamail.conv.Conv
+import emil.javamail.internal.JavaMailConnection
+import emil.{MailFolder, MailOp}
+import jakarta.mail.Folder
+
+object ListFoldersRecursive {
+  def apply[F[_]: Sync](parent: Option[MailFolder])(implicit
+      c: Conv[Folder, MailFolder]
+  ): MailOp[F, JavaMailConnection, Vector[MailFolder]] =
+    MailOp(conn =>
+      Sync[F].blocking {
+        listRecursive(
+          parent
+            .map(pf => conn.store.getFolder(pf.id))
+            .getOrElse(conn.store.getDefaultFolder)
+        )
+          .map(c.convert)
+          .toVector
+      }
+    )
+
+  private def listRecursive(
+      folder: Folder,
+      folderList: List[Folder] = Nil
```

Comment:
Hm… it seems the `folder` is never added to the result? So "INBOX" doesn't belong to the result - or the parent folder? Then `folderList` is unused, the method is not called with other than `Nil`.

It would be nice, if it would be tail-recursive, wdyt? Probably not so important in practice, but I tend to prefer it in general as it is usually not harder to write/read, imho, and so we get stack safety from scala.

Reply:
You are right about the `folderList` parameter, I was a bit tired writing this, sorry.

Not including the parent folder is intended, maybe renaming to `ListSubFolders` would be more appropriate?

How would someone transform this into a tail-recursive method? Maybe I just can't wrap my head around it, but it seems to be hard because it's a tree like structure - I'm open for ideas!

Reply:
> Not including the parent folder is intended, maybe renaming to `ListSubFolders` would be more appropriate?

Ah ok - I looked at the example in the description, it seems to be included there. Maybe we can return a `NonEmptyList` with the input folder as first element? The user can then decide whether to use it or not. I thought it may be convenient when returning the INBOX when giving a `None` as input.

Regarding the recursive thing. I thought maybe something like this could work:
``` scala
@annotation.tailrec
def flatten(folder: Vector[Folder], result: Vector[Folder] = Vector.empty): Vector[Folder] = 
  if (folder.isEmpty) result
  else flatten(folder.flatMap(_.list().tovector), folder ++ result)

// flatten(Vector(parent)) 
```
(modulo all the syntax errors) But I may just miss something here! can totally relate to the tiredness:)